### PR TITLE
Add QA script

### PR DESCRIPTION
**Pull request by maxjacobson**
The main difference from the other one is that this engine does require a configuration file.

Haven't enabled this for CI yet (there's a separate issue tracking that)

### CODE REVIEW

**Comment by maxjacobson** [comment]
Please let me know if I targeted the wrong branch here

File: qa/config.template.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "config": {
+    "languages": {
+      "%s": {
+        "mass_threshold": 20
```
Comment (dblandin) [inline qa/config.template.json]: Should we be overriding the mass_threshold here?

Comment (maxjacobson) [reply qa/config.template.json]: Nope, I don't think we need to. Updated

File: qa/bin/clone_and_test_examples [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  else
+    LANGS[0]="ruby"
+    LANGS[1]="python"
+    LANGS[2]="php"
+    LANGS[3]="java"
+    LANGS[4]="javascript"
+  fi
+
+  for lang in ${LANGS[*]}; do
+    printf "$(cat "$config_template_path")" "$lang" > "$config_path"
```
Comment (wfleming) [inline qa/bin/clone_and_test_examples]: nice.

**Pull request merged**